doc/source/qibo.rst

```diff
@@ -376,4 +385,17 @@ The user can switch backends using
     qibo.set_backend("matmuleinsum")
 
 before creating any circuits or gates. The default backend is ``"custom"`` and
-uses the custom Tensorflow operators.
+uses the custom Tensorflow operators. One can switch to a numpy backend using
+the same approach:
+
+.. code-block::  python
+
+    import qibo
+    qibo.set_backend("numpy_defaulteinsum")
```

Comment:
As discussed in `__init__.py`, I think a clear separation between backend and computational engine should be cleaner, and possibly avoid misleading information.

For example, if I follow this instruction, I get this output (which may be very confusing):
```python
import qibo
qibo.set_backend("numpy_defaulteinsum")
print(qibo.get_backend()) 
# prints: defaulteinsum

from qibo.config import BACKEND_NAME
print(BACKEND_NAME)
# prints: tensorflow instead of numpy

from qibo import K
print(K)
# prints: qibo.backends.tensorflow.TensorflowBackend instead of Numpy
```

Reply:
I fixed these issues temporarily by completely removing the `BACKEND_NAME` variable as it is not necessarily needed now and defining the `__str__` method for `K`, so that `print(K)` is consistent with the current backend and `get_backend()`. The final implementation of this depends on which approach we will use for the backend switcher.

Reply:
Thanks.